app.rb

```diff
@@ -0,0 +1,76 @@
+require 'sinatra/base'
+require 'sinatra/flash'
+
+require_relative 'data_mapper_setup.rb'
+require_relative 'lib/peep'
+require_relative 'lib/user'
+
+class ChitterApp < Sinatra::Base
+
+  setup_databases
+
+  enable :sessions
+  register Sinatra::Flash
+
+  get '/' do
+    'Hello world'
+  end
+
+  post '/users' do
+    string = "Sorry, that email or username has already been used."
+    unless (user = User.make(email: params["email"], password: params["password"], username: params["username"], name: params["name"]))
+      flash[:error] = string
+      redirect '/users/new'
+    else
+      User.sign_in(username: params["username"], password: params["password"])
+      redirect "/users/#{user.id}"
+    end
+  end
+
+  get '/users/new' do
+    erb :sign_up
+  end
+
+  get '/users/sign_in' do
+    erb :sign_in
+  end
+
+  post '/users/sign_in' do
+    string = "Sorry, either your username or password were incorrect"
+    unless (user = User.sign_in(username: params["username"], password: params["password"]))
+      flash[:error] = string
+      redirect '/users/sign_in'
+    else
+      redirect "/users/#{user.id}"
+    end
+  end
+
+  get '/users/sign_out' do
+    User.sign_out
+    redirect 'peeps'
+  end
+
+  get '/users/:id' do
+    @user = User.get(params[:id])
+    erb :users
+  end
+
+  get '/peeps' do
+    @current_user = User.current_user
+    @user_class = User
```

Comment:
I'm not sure why you need line 60.

Reply:
Ah, I see now - you are calling `@user_class.get(peep.user_id).username` in your view. This is generally bad practice, for two reasons:

1) It's a lot clearer when you define all the vars for view in advance of rendering the view, as it means that another developer can see exactly what information is needed by the view in one place
2) Fetching the data from User actually involves a call to the database. You are making multiple calls to user_class.get(peep.user_id) in your view, which means multiple calls to the database, which adds load to your system. Instead, if you have a relationship, you should be able to do peep.user.username & peep.user.name. This will result in fewer calls to the database. To understand this fully, it's worth reading about lazy loading, eager loading, and how Datamapper uses strategic eager loading to reduce database load.